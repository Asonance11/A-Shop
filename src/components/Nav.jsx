import React from 'react';
import { Link } from 'react-router-dom';

const Nav = () => {
	return (
		<header>
			<h1>
				<Link to="/">Painty</Link>
			</h1>

			<ul>
				<li>
					<Link to="/shop">Shop </Link>
				</li>
				<li>
					<Link to="/cart">Cart </Link>
				</li>
			</ul>
		</header>
	);
};

export default Nav;
