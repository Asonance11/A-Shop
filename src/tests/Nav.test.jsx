import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { describe, it } from 'vitest';
import Nav from '../components/Nav';

describe('Nav component', () => {
	it('renders Nav correctly', () => {
		const { container } = render(<Nav />, { wrapper: MemoryRouter });
		expect(container).toMatchSnapshot();
	});
});
