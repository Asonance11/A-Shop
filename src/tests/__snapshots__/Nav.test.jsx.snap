// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Nav component > renders Nav correctly 1`] = `
<div>
  <header
    class="flex justify-between items-center w-full py-4 px-8 bg-darkGrey fixed"
  >
    <h1
      class="font-serif italic font-semibold text-transparent text-5xl bg-clip-text bg-gradient-to-r from-forestGreen via-adventurousYellow to-foggyBlue"
    >
      <a
        href="/"
      >
        Painty
      </a>
    </h1>
    <ul
      class="flex items-center gap-6 text-white font-sans"
    >
      <li
        class="text-xl font-medium hover:underline"
      >
        <a
          href="/shop"
        >
          Shop 
        </a>
      </li>
      <li
        class="text-xl font-medium hover:underline"
      >
        <a
          href="/cart"
        >
          Cart 
        </a>
      </li>
    </ul>
  </header>
</div>
`;
